express-api/src/typeorm/Migrations/1717629146741-MoveMetadataToTables.ts

```diff
@@ -0,0 +1,165 @@
+import { In, MigrationInterface, QueryRunner } from 'typeorm';
+import { TimestampType } from '../Entities/TimestampType';
+import { ProjectStatus } from '../Entities/ProjectStatus';
+import { NoteType } from '../Entities/NoteType';
+import { MonetaryType } from '../Entities/MonetaryType';
+import { Task } from '../Entities/Task';
+import { User } from '../Entities/User';
+
+export class MoveMetadataToTables1717629146741 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    const systemUser = await queryRunner.manager.findOne(User, { where: { Username: 'system' } });
+    const systemId = systemUser.Id;
+    const result = await queryRunner.manager.save(ProjectStatus, {
+      Id: 44, //For some reason there is no sequence on project_status
+      Name: 'Close Out',
+      Code: 'CLS-OUT',
+      Description: 'Status related to filling out the close out form.',
+      IsMilestone: true,
+      IsTerminal: true,
+      Route: '/place/holder',
+      CreatedById: systemId,
+    });
+    await queryRunner.manager.insert(TimestampType, [
+      { Name: 'AdjustedOn', IsOptional: true, StatusId: result.Id, CreatedById: systemId }, //ProjectCloseOut
+      { Name: 'ApprovedForSplOn', IsOptional: true, StatusId: 21, CreatedById: systemId }, //ProjectSPLApproval, ProjectERPComplete
+      {
+        Name: 'ClearanceNotificationSentOn',
+        IsOptional: true,
+        StatusId: 14,
+        CreatedById: systemId,
+      }, //ProjectERPComplete
+      { Name: 'DisposedOn', IsOptional: true, StatusId: 32, CreatedById: systemId }, //ProjectERPDisposed, ProjectCloseOut
+      { Name: 'ExemptionApprovedOn', IsOptional: true, StatusId: 15, CreatedById: systemId }, //ProjectERPExemption
+      { Name: 'FinalFormSignedOn', IsOptional: true, StatusId: result.Id, CreatedById: systemId }, //ProjectCloseOut
+      { Name: 'InterestedReceivedOn', IsOptional: true, StatusId: 14, CreatedById: systemId }, //ProjectERPComplete
+      { Name: 'OfferAcceptedOn', IsOptional: true, StatusId: 32, CreatedById: systemId }, //ProjectERPDisposed, ProjectNotSPL, ProjectSPLContractInPlace
+      { Name: 'RemovalFromSplApprovedOn', IsOptional: true, StatusId: 21, CreatedById: systemId }, //ProjectSPLApproval
+      { Name: 'RemovalFromSplRequestOn', IsOptional: true, StatusId: 21, CreatedById: systemId }, //ProjectSPLApproval
+      { Name: 'RequestForSplReceivedOn', IsOptional: true, StatusId: 14, CreatedById: systemId }, //ProjectERPComplete, ProjectSPLApproval
+      { Name: 'TransferredWithinGreOn', IsOptional: true, StatusId: 14, CreatedById: systemId }, //ProjectERPComplete, ProjectNotSPL
```

Comment:
Why not status 20? "Transferred within the GRE"?

Reply:
Yeah this could make the most sense... Another one that is referenced more than once. But I think it's only showed conditionally in those other places.